docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
+Patients by selected criteria.
+
+#### `Doctor`
+
+![DoctorUML](diagrams/Doctor.png)
+
+The Doctor class holds several attributes that are exposed via getter methods. To instantiate the class, all the
+attributes must be given to the constructor method. There is a method to edit the attribute data that can be called
+whenever user types in the correct command.
+
+#### `Doctor List`
+
+![DoctorListUML](diagrams/DoctorList.png)
+
+The DoctorList class contains private lists of Doctors and one list for searching. It has several
+public methods that allows the user to get information regarding the list, view the list as well as search for specific
+Doctors by selected criteria.
+
+--------------------------------------------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Implementation**
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### View appointments with selected criteria
+This feature is a method within `AppointmentList`. `AppointmentList` contains an arraylist
+of `Appointment` as a private element. `AppointmentList` methods invoked interacts with this
+list, possibly making changes in the process.
+
+Currently, `AppointmentList` has the following methods:
+* `AppointmentList#add` -- Appends a new `Appointment` to the list.
+* `AppointmentList#remove` -- Removes an entry from the list by appointment id.
+* `AppointmentList#edit` -- Edits a existing entry from the list by.
+* `AppointmentList#view` -- Text display of all or selected appointments in the list.
+* `AppointmentList#find` -- Find selected appointments using the criteria given.
+
+The methods are exposed in the `Manager#runLoop` method where user input is parsed. A `ViewAppointmentCommand` will
+be returned if the user input passes the validation by `Validator`. Else, a `HalpmiException` will be thrown
+indicating missing parameters.
+
+Below is an example describing the behaviour of the `find` feature.
+
+#### Design considerations:
+
+---------------------------------------------------------------------------------------------------------------
 ## Product scope
-### Target user profile
 
-{Describe the target user profile}
+###Target user profile:
+
+* administrator in clinic(s)
+* prefer desktop apps over other types
+* can type fast
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
+
+###Value proposition:
 
-### Value proposition
+Manage core clinic related data faster than using mouse or GUI.
+Ensure each data type conforms to certain standards with in-built validations.
 
-{Describe the value proposition: what problem does it solve?}
+### User stories
 
-## User Stories
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
```

Comment:
I think the formatting of this legend can be neater? Overall great idea nonetheless!